movement/io/load_bboxes.py

```diff
@@ -0,0 +1,520 @@
+            if re.search(pattern, f)
+            else np.nan
+            for f in df["filename"]
+        ]
+
+        frame_array = np.array(list_frame_numbers).reshape(-1, 1)
+
+    return frame_array
+
+
+def _via_attribute_column_to_numpy(
+    df: pd.DataFrame,
+    via_column_name: str,
+    list_keys: list[str],
+    cast_fn: Callable = float,
+) -> np.ndarray:
+    """Convert values from VIA attribute-type column to a numpy array.
+
+    In the VIA tracks file, the attribute-type columns are the columns
+    whose name includes the word `attributes` (i.e. `file_attributes`,
+    `region_shape_attributes` or `region_attributes`). These columns hold
+    dictionary data.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The pandas DataFrame containing the data from the VIA file.
+        This is the dataframe obtained from running
+        `df = pd.read_csv(file_path, sep=",", header=0)`.
+    via_column_name : str
+        The name of a column in the VIA file whose values are literal
+        dictionaries (i.e. `file_attributes`, `region_shape_attributes`
+        or `region_attributes`).
+    list_keys : list[str]
+        The list of keys whose values we want to extract from the literal
+        dictionaries in the `via_column_name` column.
+    cast_fn : type, optional
+        The type function to cast the values to. By default `float`.
+
+    Returns
+    -------
+    np.ndarray
+        A numpy array holding the extracted values. The rows (axis=0) are the
+        rows of the input dataframe. The columns (axis=1) follow the order of
+        the `list_attributes`.  Arrays will have at least 1 column
+        (no 0-rank numpy arrays).
+
+    """
+    list_bbox_attr = []
+    for _, row in df.iterrows():
+        row_dict_data = ast.literal_eval(row[via_column_name])
+        list_bbox_attr.append(
+            tuple(cast_fn(row_dict_data[reg]) for reg in list_keys)
+        )
+
+    bbox_attr_array = np.array(list_bbox_attr)
+
+    return bbox_attr_array
+
+
+def _ds_from_valid_data(data: ValidBboxesDataset) -> xr.Dataset:
+    """Convert a validated bboxes dataset to an xarray Dataset.
+
+    Parameters
+    ----------
+    data : movement.io.tracks_validators.ValidPosesDataset
+        The validated data object.
+
+    Returns
+    -------
+    bounding boxes dataset containing the boxes tracks,
+        boxes shapes, confidence scores and associated metadata.
+
+    """
+    # Create the time coordinate
+    time_coords = data.frame_array
+    time_unit = "frames"
+    # if fps is provided: time_coords is expressed in seconds.
+    if data.fps is not None:
+        # Compute frames from the start (first frame is frame 0).
+        # Ignoring type error because `data.frame_array` is not None after
+        # ValidBboxesDataset.__attrs_post_init__()
+        time_coords = np.arange(data.frame_array.shape[0], dtype=int)  # type: ignore
+        time_coords = time_coords / data.fps
+        time_unit = "seconds"
```

Comment:
I'm confused about the need for and the nature of `frame_array`.
We may have discussed this but I forgot and would like to discuss it again.
Is it because the tracking data may not correspond to the entire source video, but instead to a subset of it? For example if the video is 1000 frames long, and only the 2nd half is tracked, then we would have `frame_array=np.array([500, 501, 502, ..., 999])`?

If my assumption is correct and the array consists of contiguous numbers, would it not just suffice to supply a `frame_offset` (i.e. a single integer value) and then calculate the length from the length of the `position` array? In that case the `ValidBboxesDataset` can be made to accept `frame_offset` instead of `frame_array`.

Moreover, there is a bit of an inconsistency in your current implementation, because frame numbers are relative to the video, while seconds are relative to the start of tracking. Namely, if a user supplied the `fps`, then time coordinates will start from `0` i.e. from the beginning of tracking, whereas is `fps` is missing they will start from the first tracked frame (`frame_array[0]` or `frame_offset`).

If my assumption is wrong and the `frame_array` can be non-contiguous (e.g. `[500, 505, 770]`), then the way you translate frames to time here also wouldn't make sense, because for this example array and `fps=40` we would get `time_coords=[0, 0.025, 0.05]`. Non-contiguous frames are useful for labels, so were you perhaps anticipating labelled or non-tracked data?

In general, `movement` currently can't really handle non-contiguous frames or non-constant inter-frame intervals (assumes fixed `fps`). For a proper handling of time and frames, we would first have to solve https://github.com/neuroinformatics-unit/movement/issues/117. I'm happy to brainstorm this together. I thinks that issue should be prioritised, because it's also essential for loading labelled data.

Reply:
We talked about this offline but I thought it would be good to make a note here to keep a record of our conclusions.

> Is it because the tracking data may not correspond to the entire source video, but instead to a subset of it? For example if the video is 1000 frames long, and only the 2nd half is tracked, then we would have frame_array=np.array([500, 501, 502, ..., 999])?

Yes. Right now `frame_array` is optional, so if we don't pass it the frames will be numbered from 0 to `n`.

> would it not just suffice to supply a `frame_offset`?

Yes, that would be an alternative. In our discussion we weren't sure if one or other option was significantly advantageous. We thought having a `frame_array` may make it easier to adapt the code to load manually labelled frames, so we decided to stick with that approach for now.

> Moreover, there is a bit of an inconsistency in your current implementation, because frame numbers are relative to the video, while seconds are relative to the start of tracking

Yes, that is correct. I thought that was more intuitive, but it is also inconsistent. We agreed to make the time dimension to always have the same origin, whether it is expressed in frames or seconds. So if the video is only tracked from frame 500 to 999, and we would like to express time in seconds, we assume the origin of time (time = 0 seconds) is when frame 0 is captured.

> If my assumption is wrong and the frame_array can be non-contiguous...

We decided to add a validator check that ensures the frames passed via `frame_array` are contiguous